eng/pipelines/templates/jobs/vmr-build.yml

```diff
@@ -180,146 +195,156 @@ jobs:
-      find src/ -type f -name "*.binlog" -exec cp {} --parents -t ${targetFolder} \;
-      find src/ -type f -name "*.log" -exec cp {} --parents -t ${targetFolder} \;
-      find test/ -type f -name "*.binlog" -exec cp {} --parents -t ${targetFolder} \;
-      find test/ -type f -name "Updated*.diff" -exec cp {} --parents -t ${targetFolder} \;
-      find test/ -type f -name "Updated*.txt" -exec cp {} --parents -t ${targetFolder} \;
-    displayName: Prepare BuildLogs staging directory
-    continueOnError: true
-    condition: succeededOrFailed()
+        if [[ '${{ parameters.buildFromArchive }}' == 'True' ]]; then
+          customBuildArgs="$customBuildArgs --source-repository https://github.com/dotnet/dotnet"
+          customBuildArgs="$customBuildArgs --source-version $(git -C "${{ parameters.vmrPath }}" rev-parse HEAD)"
+        fi
+
+        if [[ '${{ parameters.buildSourceOnly }}' == 'True' ]]; then
+          customBuildArgs="$customBuildArgs --source-only"
+        fi
+
+        if [[ '${{ parameters.useMonoRuntime }}' == 'True' ]]; then
+          customBuildArgs="$customBuildArgs --use-mono-runtime"
+        fi
+
+        if [[ -n "${{ parameters.container }}" ]]; then
+          useDocker=true
+        fi
+
+        if [[ ! -z '${{ parameters.targetRid }}' ]]; then
+          extraBuildProperties="--"
+          if [[ '${{ parameters.useMonoRuntime }}' == 'True' ]]; then
+              customEnvVars="$customEnvVars CROSSCOMPILE=1"
+          fi
+          extraBuildProperties="$extraBuildProperties /p:PortableBuild=true /p:DotNetBuildVertical=true /p:CrossBuild=true"
+        fi
+
+        if [[ ! -z '${{ parameters.crossRootFs }}' ]]; then
+          customEnvVars="$customEnvVars ROOTFS_DIR=${{ parameters.crossRootFs}}"
+        fi
+
+        if [[ ! -z '${{ parameters.targetRid }}' ]]; then
+          extraBuildProperties="$extraBuildProperties /p:OverrideTargetRid=${{ parameters.targetRid }}"
+        fi
+
+        # Only use Docker stuff on Linux
+        if [[ "$useDocker" == "true" ]]; then
+          customDockerRunArgs=""
+          for envVar in $customEnvVars; do
+            customDockerRunArgs="$customDockerRunArgs -e $envVar"
+          done
+          
+          if [[ '${{ parameters.runOnline }}' == 'False' ]]; then
+            customDockerRunArgs="$customDockerRunArgs --network none"
+          fi
+
+          docker run --rm -v "$(sourcesPath):/vmr" -w /vmr $customDockerRunArgs ${{ parameters.container }} ./build.sh --clean-while-building $(additionalBuildArgs) $customBuildArgs $extraBuildProperties
+        else
+          for envVar in $customEnvVars; do
+            customEnvVarsWithBashSyntax="$customEnvVarsWithBashSyntax export $envVar;"
+          done
+          cd $(sourcesPath)
+          eval $customEnvVarsWithBashSyntax
+          ./build.sh --source-only --clean-while-building $(additionalBuildArgs) $customBuildArgs $extraBuildProperties
```

Comment:
--source-only doesn't seem right here.  It should be part of customBuildArgs.

Please consider defining a single variable for the build cmd/args and share it between the docker/non-docker code paths.

Reply:
Shouldn't `--source-only` be used both here _and_ in the container scenario? That's how it was previously defined. Yes, I agree with combining all these args into a single variable that gets shared by both commands.